docs/architecture/adr-043-nft-module.md

```diff
@@ -0,0 +1,413 @@
+  We envision that identifiers can accommodate mint and transfer actions.
+  The `id` is also the primary index for storing NFTs.
+
+  ```
+  {type}/{id} --> NFT (bytes)
+  ```
+
+- The `uri` points to an immutable off-chain resource containing more attributes about his NFT.
+
+- The `data` is mutable field and allows attaching special information to the NFT, for example:
+
+  - metadata such as the title of the work and URI.
+  - immutable data and parameters (such actual NFT data, hash or seed for generators).
+  - mutable data and parameters that change when transferring or when certain criteria are met (such as provenance).
+
+  This ADR doesn't specify values that this field can take; however, best practices recommend upper-level NFT modules clearly specify their contents.
+  Although the value of this field doesn't provide the additional context required to manage NFT records, which means that the field can technically be removed from the specification,
+  the field's existence allows basic informational/UI functionality.
+
+- The ownership of nft is controlled by the `x/bank` module and the `metadata` part will be converted to `banktypes.Metadata` is stored in the `x/bank` module.
+
+### `Msg` Service
+
+```protobuf
+service Msg {
+  rpc Issue(MsgIssue) returns (MsgIssueResponse);
+  rpc Mint(MsgMint)   returns (MsgMintResponse);
+  rpc Edit(MsgEdit)   returns (MsgEditResponse);
+  rpc Send(MsgSend)   returns (MsgMsgSendResponse);
+  rpc Burn(MsgBurn)   returns (MsgBurnResponse);
+}
+
+message MsgIssue {
+  cosmos.nft.v1beta1.Metadata metadata = 1;
+  string issuer                        = 2;
+}
+message MsgIssueResponse {}
+
+message MsgMint {
+  cosmos.nft.v1beta1.NFT nft = 1;
+  string minter              = 2;
+}
+message MsgMintResponse {}
+
+message MsgEdit {
+  cosmos.nft.v1beta1.NFT nft = 1;
+  string editor              = 2;
+}
+message MsgEditResponse {}
+
+message MsgSend {
+  string type     = 1;
+  string id       = 2;
+  string sender   = 3;
+  string reveiver = 4;
+}
+message MsgSendResponse {}
+
+message MsgBurn {
+  string type      = 1;
+  string id        = 2;
+  string destroyer = 3;
+}
+message MsgBurnResponse {}
+```
+
+`MsgIssue` can be used to issue an NFT type/class, just like deploying an ERC721 contract on Ethereum.
+
+`MsgMint` allows users to create new NFTs for a given type.
+
+`MsgEdit` allows users to edit/update their NFTs.
+
+`MsgSend` can be used to transfer the ownership of an NFT to another address.
+**Note**: we could use `x/bank` to handle NFT transfer directly and do without this service. It's only for the sake of completeness of an ERC721 compatible API that we may choose to keep this service.
```

Comment:
I forgot to add a comment: I think we don't need `MsgSend` - `x/bank` MsgSend works perfectly well.